python-sdk/src/astro/utils/serializer.py

```diff
@@ -0,0 +1,96 @@
+from __future__ import annotations
+
+import json
+import logging
+import pickle
+from json import JSONDecodeError
+from pickle import UnpicklingError
+from typing import Any
+
+import pandas
+from astro.files import File
+from astro.sql.table import Table, TempTable
+from astro.utils.dataframe import convert_to_file
+
+log = logging.getLogger("astro.utils.serializer")
+
+
+def serialize(obj: Table | File | Any) -> dict | Any:
+    """
+    Serialize astro SDK objects (tables, files and dataframes) into json safe dictionary
+
+    :param obj:
+    :return:
+    """
+    if isinstance(obj, (Table, TempTable)):
+        return obj.to_json()
+    elif isinstance(obj, pandas.DataFrame):
+        file = convert_to_file(obj)
+        return serialize(file)
+    elif isinstance(obj, File):
+        return obj.to_json()
+    elif isinstance(obj, list):
+        return [serialize(o) for o in obj]
+    elif isinstance(obj, str):
+        return {"class": "string", "value": obj}
+    else:
+        return _attempt_to_serialize_unknown_object(obj)
+
+
+def _attempt_to_serialize_unknown_object(obj: object):
+    try:
+        return json.dumps(obj)
+    except TypeError:
+        return pickle.dumps(obj).hex()
+
+
+def _is_serialized_astro_object(obj) -> bool:
+    return bool(obj.get("class") and obj["class"] in ["Table", "File", "string"])
+
+
+def deserialize(obj: dict | str | list) -> Table | File | Any:
+    """
+    Deserialize json dictionaries into astro SDK objects (tables, files, dataframes, etc.)
+
+    :param obj:
+    :return:
+    """
+    if isinstance(obj, (list, tuple)):
+        return [deserialize(o) for o in obj]
+    if isinstance(obj, dict) and _is_serialized_astro_object(obj):
+        if obj["class"] == "Table":
+            log.debug("Found table dictionary %s, will attempt to deserialize", obj)
+            return Table.from_json(obj)
+        elif obj["class"] == "File":
+            log.debug("Found file dictionary %s, will attempt to deserialize", obj)
+            return _deserialize_file(obj)
+        else:
+            return obj["value"]
+    elif isinstance(obj, str):
+        log.debug("Found string, will attempt to deserialize")
+        return _attempt_to_deser_unknown_object(obj)
+    else:
+        return obj
```

Comment:
I don't understand what you're trying to do here?

Reply:
Removing the `elif` and `else`'s as if you have _if-return_'s you don't need them.

Reply:
With multiple returns what you want to do is essentially a row of shortcuts. A shortcut will be taken if the condition is met. In the end in case of no shortcuts taken you just return what you would return without shortcuts. (I don't know if this was helpful or not 😅)

It is my personal preference, but feel free to keep it as it is :)

Reply:
I'm gonna keep it as is to prevent any unwanted actions if a return statement is removed :)